development/predict_3d_model_with_lucchi.py

```diff
@@ -0,0 +1,191 @@
+from torch_em.util import load_model
+from torch_em.transform.raw import normalize
+from torch_em.util.prediction import predict_with_halo
+
+from micro_sam import util
+from micro_sam.evaluation.inference import _run_inference_with_iterative_prompting_for_image
+
+from segment_anything import SamPredictor
+
+from micro_sam.models.sam_3d_wrapper import get_sam_3d_model
+from typing import List, Union, Dict, Optional, Tuple
+
+
+class RawTrafoFor3dInputs:
+    def _normalize_inputs(self, raw):
+        raw = normalize(raw)
+        raw = raw * 255
+        return raw
+
+    def _set_channels_for_inputs(self, raw):
+        raw = np.stack([raw] * 3, axis=0)
+        return raw
+
+    def __call__(self, raw):
+        raw = self._normalize_inputs(raw)
+        raw = self._set_channels_for_inputs(raw)
+        return raw
+
+
+def _run_semantic_segmentation_for_image_3d(
+    model: torch.nn.Module,
+    image: np.ndarray,
+    prediction_path: Union[os.PathLike, str],
+    patch_shape: Tuple[int, int, int],
+    halo: Tuple[int, int, int],
+):
+    device = next(model.parameters()).device
+    block_shape = tuple(bs - 2 * ha for bs, ha in zip(patch_shape, halo))
+
+    def preprocess(x):
+        x = 255 * normalize(x)
+        x = np.stack([x] * 3)
+        return x
+
+    def prediction_function(net, inp):
+        # Note: we have two singleton axis in front here, I am not quite sure why.
+        # Both need to be removed to be compatible with the SAM network.
+        batched_input = [{
+            "image": inp[0, 0], "original_size": inp.shape[-2:]
+        }]
+        masks = net(batched_input, multimask_output=True)[0]["masks"]
+        masks = torch.argmax(masks, dim=1)
+        return masks
+
+    # num_classes = model.sam_model.mask_decoder.num_multimask_outputs
+    image_size = patch_shape[-1]
+    output = np.zeros(image.shape, dtype="float32")
+    predict_with_halo(
+        image, model, gpu_ids=[device],
+        block_shape=block_shape, halo=halo,
+        preprocess=preprocess, output=output,
+        prediction_function=prediction_function
+    )
+
+    # save the segmentations
+    imageio.imwrite(prediction_path, output, compression="zlib")
+
+
+def run_semantic_segmentation_3d(
+    model: torch.nn.Module,
+    image_paths: List[Union[str, os.PathLike]],
+    prediction_dir: Union[str, os.PathLike],
+    semantic_class_map: Dict[str, int],
+    patch_shape: Tuple[int, int, int] = (32, 512, 512),
+    halo: Tuple[int, int, int] = (6, 64, 64),
+    image_key: Optional[str] = None,
+    is_multiclass: bool = False,
+):
+    """
+    """
+    for image_path in tqdm(image_paths, desc="Run inference for semantic segmentation with all images"):
+        image_name = os.path.basename(image_path)
+
+        assert os.path.exists(image_path), image_path
+
+        # Perform segmentation only on the semantic class
+        for i, (semantic_class_name, _) in enumerate(semantic_class_map.items()):
```

Comment:
I would suggest to remove this part already, it doesn't make sense in the context here.